rustls/src/msgs/codec.rs

```diff
@@ -120,6 +120,7 @@ impl u24 {
     }
 }
 
+#[allow(clippy::from_over_into)]
```

Comment:
It seems like we should follow this PR up with a new one that does what the lint suggests. I think these are only written the way they are because of the old coherence rules.

Reply:
In my opinion, conceptually it does not make sense to have a conversion from `usize` -> `u24`.

Reply:
I think the `from_over_into` lint is telling you to `impl From<u24> for usize` instead of of `impl Into<usize> for u24`; i.e. the same conversion you're currently doing, but in terms of `From`.

Reply:
Sure, but the reason given is that `From` impls have a blanket impl for the converse `Into` impl, which in this case seems undesirable to me.

Reply:
> Sure, but the reason given is that From impls have a blanket impl for the converse Into impl, which in this case seems undesirable to me.

No, it won't automatically generate the "converse." It will instead generate this exact `impl Into<usize> for u24`.

Basically, it's saying "Hey, it looks like you are implementing `Into<usize> for u24`  because the old coherence rules won't let you write `From<u24> for usize`, but the new coherence rules do let you do that, and that's more idiomatic."

Basically we should almost never implement `Into` directly in any case where it's possible to implement the equivalent `From` instead.

Reply:
Hmm, you're right, of course. Implemented this in #631.